lib/Sema/DerivedConformanceDistributedActor.cpp

```diff
@@ -94,6 +100,321 @@ static FuncDecl *deriveDistributedActor_resolve(DerivedConformance &derived) {
+
+  // Statements
+  SmallVector<ASTNode, 8> stmts;
+
+  // --- `let m = metatype as! SerializationRequirement.Type`
+  VarDecl *metatypeVar =
+      new (C) VarDecl(/*isStatic=*/false, VarDecl::Introducer::Let, sloc,
+                      C.getIdentifier("m"), func);
+  {
+    metatypeVar->setImplicit();
+    metatypeVar->setSynthesized();
+
+    // metatype as! <<concrete SerializationRequirement.Type>>
+    auto metatypeRef =
+        new (C) DeclRefExpr(ConcreteDeclRef(metatypeParam), dloc, implicit);
+    auto metatypeSRCastExpr = ForcedCheckedCastExpr::createImplicit(
+        C, metatypeRef, serializationRequirementMetaTypeTy);
+
+    auto metatypePattern = NamedPattern::createImplicit(C, metatypeVar);
+    auto metatypePB = PatternBindingDecl::createImplicit(
+        C, swift::StaticSpellingKind::None, metatypePattern,
+        /*expr=*/metatypeSRCastExpr, func);
+
+    stmts.push_back(metatypePB);
+    stmts.push_back(metatypeVar);
+  }
+
+  // --- Declare the local function `doInvokeOnReturn`...
+  FuncDecl *doInvokeOnReturnFunc = createLocalFunc_doInvokeOnReturn(
+      C, func,
+      nominal, handlerParam, resultBufParam);
+  stmts.push_back(doInvokeOnReturnFunc);
+
+  // --- try await _openExistential(metatypeVar, do: <<doInvokeLocalFunc>>)
+  {
+    auto openExistentialBaseIdent = C.getIdentifier("_openExistential");
+    auto doIdent = C.getIdentifier("do");
+
+    auto openExArgs = ArgumentList::createImplicit(
+        C, {
+               Argument(sloc, Identifier(),
+                        new (C) DeclRefExpr(ConcreteDeclRef(metatypeVar), dloc,
+                                            implicit)),
+               Argument(sloc, doIdent,
+                        new (C) DeclRefExpr(ConcreteDeclRef(doInvokeOnReturnFunc),
+                                            dloc, implicit)),
+           });
+    Expr *tryAwaitDoOpenExistential =
+        CallExpr::createImplicit(C,
+                                 UnresolvedDeclRefExpr::createImplicit(
+                                     C, openExistentialBaseIdent),
+                                 openExArgs);
+
+    tryAwaitDoOpenExistential =
+        AwaitExpr::createImplicit(C, sloc, tryAwaitDoOpenExistential);
+    tryAwaitDoOpenExistential =
+        TryExpr::createImplicit(C, sloc, tryAwaitDoOpenExistential);
+
+    stmts.push_back(tryAwaitDoOpenExistential);
+  }
+
+  auto body = BraceStmt::create(C, sloc, {stmts}, sloc, implicit);
+  return {body, /*isTypeChecked=*/false};
+}
+
+/// Synthesizes the
+///
+/// \verbatim
+/// static func invokeHandlerOnReturn(
+////    handler: ResultHandler,
+////    resultBuffer: UnsafeRawPointer,
+////    metatype _metatype: Any.Type
+////  ) async throws
+/// \endverbatim
+static FuncDecl *deriveDistributedActorSystem_invokeHandlerOnReturn(
+    DerivedConformance &derived) {
+  auto system = derived.Nominal;
+  auto &C = system->getASTContext();
+
+  auto mkParam = [&](Identifier argName, Identifier paramName,
```

Comment:
Looks like this is where newly added `ParamDecl::createImplicit` could come handy.